### PR TITLE
chore: Bump k8s to 1.29

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

Bump the following to support k8s v1.29 :
- k8s
- k8s-utils
- skv2
- solo-kit
- solo-apis

## CI changes
- CI now runs on a kind k8s 1.29.2 cluster
- Ingress will be deprecated on 1.17+. Ref: https://solo-io-corp.slack.com/archives/G01EERAK3KJ/p1716389614777799
- Add support for on-demand regression and performance tests. Copied over from [Enterprise](https://github.com/solo-io/solo-projects/pull/5869)

# Context
https://github.com/solo-io/gloo/issues/8978

## Testing steps
CI passing
Nightly tests [passing](https://github.com/solo-io/gloo/actions/runs/9210119536/job/25336344692) on 1.25 and 1.29

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/cli/pkg/cmd/install/knative_test_yaml.yaml [hunk 1]
```diff
@@ -4617,7 +4617,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
-        image: k8s.gcr.io/elasticsearch:v5.6.4
+        image: registry.k8s.io/elasticsearch:v5.6.4
```
Comment (davidjumani) [inline projects/gloo/cli/pkg/cmd/install/knative_test_yaml.yaml:4620]: Ref: https://kubernetes.io/blog/2023/11/16/kubernetes-1-29-upcoming-changes/#a-note-about-the-k8s-gcr-io-redirect-to-registry-k8s-io

File: changelog/v1.17.0-beta33/bump-k8s-129.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+changelog:
+- type: BREAKING_CHANGE
+  description: >
+    Bump dependencies to support Kubernetes 1.29
```
Comment (bewebi) [inline changelog/v1.17.0-beta33/bump-k8s-129.yaml:4]: nit: Is the bump itself breaking?

I know we're deprecating Ingress and support for k8s 1.24, which I think constitute breaking changes and should be called out here

File: go.mod [hunk 3]
```diff
@@ -49,20 +49,22 @@ require (
...
+	github.com/solo-io/k8s-utils v0.7.2
 	github.com/solo-io/protoc-gen-ext v0.0.18
 	github.com/solo-io/protoc-gen-openapi v0.2.2
-	github.com/solo-io/skv2 v0.36.5
+	github.com/solo-io/skv2 v0.39.1
 
-	// Pinned to the latest `gloo-repo-branch` tag of solo-apis (`gloo-v1.17.0-beta25`)
-	github.com/solo-io/solo-apis v0.0.0-20240503163415-57628abb2459
-	github.com/solo-io/solo-kit v0.34.2
+	// Pinned to the latest `gloo-main` tag of solo-apis (`sa-k8s-1.29-bump`)
```
Comment (sheidkamp) [inline go.mod:57]:
```suggestion
	// Pinned to the `sa-k8s-1.29-bump` of `gloo-main` tag of solo-apis
```

Comment (sheidkamp) [reply go.mod:57]: its not going to be the "latest" tag for long, right?

File: .github/workflows/nightly-tests.yaml [hunk 4]
```diff
@@ -23,6 +23,7 @@ on:
           - main
           - v1.16.x
           - v1.15.x
+          - workflow_initiating_branch
```
Comment (bewebi) [inline .github/workflows/nightly-tests.yaml:26]: nit: Worth mentioning the "on demand" additions, which I gather are brought over from solo-io/solo-projects#5869, in the PR description

**Review by bewebi — COMMENTED** [review]
a couple nits

File: projects/gloo/cli/pkg/cmd/check/root.go [hunk 5]
```diff
@@ -382,6 +382,8 @@ func checkPods(ctx context.Context, printer printers.P, opts *options.Options) e
 				if conditionNotMet {
 					errorToPrint = fmt.Sprintf("Not all containers in pod %s in namespace %s are ready!%s", pod.Name, pod.Namespace, message)
 				}
+			case corev1.PodReadyToStartContainers:
+				// This condition was introduced in k8s 1.29. Skip it since completed jobs have Status=False for this condition
```
Comment (sheidkamp) [inline projects/gloo/cli/pkg/cmd/check/root.go:386]: Nit: add a log message here jus in case something weird happens?

**Comment by sheidkamp** [comment]
Should we update the supported versions docs as well https://github.com/solo-io/gloo/blob/bump-k8s-129/docs/content/reference/support.md#supported-versions ?

**Review by sheidkamp — COMMENTED** [review]
Also noted in a non-review comment - do we need to update https://github.com/solo-io/gloo/blob/bump-k8s-129/docs/content/reference/support.md#supported-versions ?

File: projects/gloo/cli/pkg/cmd/check/root.go (hunk 5 above)
Comment (davidjumani) [reply projects/gloo/cli/pkg/cmd/check/root.go:386]: Adding a log message here leads to test failures as the tests check for a [specific string](https://github.com/solo-io/gloo/blob/main/test/kubernetes/e2e/features/glooctl/check.go#L29)
`Checking pods... OK`
Adding log lines can lead to `Checking pods... <log message>OK`

File: go.mod [hunk 6]
```diff
@@ -54,7 +54,7 @@ require (
 	github.com/solo-io/protoc-gen-openapi v0.2.2
 	github.com/solo-io/skv2 v0.39.1
 
-	// Pinned to the latest `gloo-main` tag of solo-apis (`sa-k8s-1.29-bump`)
+	// Pinned to the `sa-k8s-1.29-bump` of `gloo-main` tag of solo-apis
```
Comment (bewebi) [inline go.mod:57]:
```suggestion
	// Pinned to the `sa-k8s-1.29-bump` tag of solo-apis on `gloo-main` branch
```

File: go.mod (hunk 3 above)
Comment (davidjumani) [reply go.mod]: Right. Updated

File: .github/workflows/nightly-tests.yaml (hunk 4 above)
Comment (davidjumani) [reply .github/workflows/nightly-tests.yaml:26]: Updated

**Review by bewebi — APPROVED** [review]

**Review by sheidkamp — APPROVED** [review]

**Comment by davidjumani** [comment]
kick bulldozer

**Pull request merged**